### ISSUE TITLE
Add/edit descriptions of existing playlists?

### ISSUE THREAD

**Issue by JVMed** [simon-weber/Unofficial-Google-Music-API#300]
It is already possible to create playlists with descriptions with 

    Webclient.create_playlist(name, description, public=False)

but would it be possible to edit/add descriptions to existing user created playlists in the future? I can edit/add descriptions on the Android Play Music app.

**Comment by simon-weber** [comment]
If it can be done from the app, then it should be pretty straightforward to add.

**Comment by JVMed** [comment]
That would be cool! It's also possible to edit the playlist title from the app: 

![screenshot](http://i.imgur.com/h6tWW94.png)

**Comment by simon-weber** [comment]
blocked on #303

**Comment by simon-weber** [comment]
request
```
PUT https://mclients.googleapis.com/sj/v1.10/playlists/<id>?alt=json&hl=en_US
{
	"creationTimestamp": "-1",
	"deleted": false,
	"description": "...",
	"id": "...",
	"lastModifiedTimestamp": "1420400606246000",
	"name": "...",
	"ownerName": "...",
	"shareToken": "...",
	"type": "USER_GENERATED"
}
```

response is the current state (same as request assuming serial requests)

**Comment by simon-weber** [comment]
unfortunately, it doesn't look like this endpoint handles patch requests. It might be easier to do this through the batch_mutate call that change_playlist_name currently uses.

**Comment by thebigmunch** [comment]
Is there a reason that change_playlist_name isn't in the documentation?

PS I quickly added a change_playlist_description method and added a description param to Mobileclient.create_playlist while on a conference call this morning : P

**Comment by thebigmunch** [comment]
Also, there doesn't seem to be any testing done for playlist descriptions. Should something be added for that? Or is it fine to leave it?

**Comment by simon-weber** [comment]
> Is there a reason that change_playlist_name isn't in the documentation?

Whoops; nope. We should add it.

**Comment by thebigmunch** [comment]
Yeah, I'll do that, too.

**Comment by thebigmunch** [comment]
@simon-weber: So, as I was checking out create_playlist, I noticed that in gmusicapi description is required in the Webclient. This is inconsistent with Google's behavior which lets you create a playlist with no description. What do you think about changing that and having the Mobileclient version follow suit?

**Comment by thebigmunch** [comment]
As I'm thinking about it more, you can also toggle a playlist as public or not. Wondering if all these should be wrapped up into one method or should I add a method for setting that separately?

**Comment by thebigmunch** [comment]
Alternatively, would you want all of the individual methods and a combined edit_playlist method?

**Comment by simon-weber** [comment]
I like to keep the client library as close to the actual calls as possible. So, if the request supports all of these changes in one call, let's expose all of them in one call too.

**Comment by JVMed** [comment]
> PS I quickly added a change_playlist_description method

Where is **change_playlist_description**, I cannot find it (maybe because I'm not that familiar with github)? 

**change_playlist_name** works fine for me

**Comment by simon-weber** [comment]
Here it is: https://github.com/simon-weber/Unofficial-Google-Music-API/blob/e7e2e3d9ee90ede66356163f2a3f752a4a34fa14/gmusicapi/clients/mobileclient.py#L301.

**Comment by simon-weber** [comment]
Whoops, I misread you. It's on @thebigmunch's fork; you either need to wait for it to be merged or use his code directly.

**Comment by JVMed** [comment]
Thanks for the quick reply Simon, but still cannot find it haha.

https://github.com/thebigmunch/Unofficial-Google-Music-API/search?utf8=%E2%9C%93&q=change_playlist_description

**Comment by simon-weber** [comment]
@thebigmunch probably just didn't push it yet.

**Comment by thebigmunch** [comment]
Yeah, I meant locally. Sorry, my back started acting up so I stopped spending so much time on coding work this week. I'll make Pull Requests for a few things today, for sure.

[Here](https://github.com/thebigmunch/Unofficial-Google-Music-API/tree/mcplaylist) is a branch that contains a few playlist-related changes. They include a fix for setting playlists as public, an optional description parameter to Mobileclient.create_playlist, and a new Mobileclient.edit_playlist method.

**Comment by JVMed** [comment]
Just tested out your Mobileclient.edit_playlist method and it works great for names, descriptions and public/private @thebigmunch! Thank you both!

**Comment by simon-weber** [comment]
Closed by b45e1113.
